broadinstitute/single_cell_portal_core
# Improve onboarding for differential expression upload (SCP-5235)

This improves in-app documentation and prominence for the "Differential expression" step in the upload wizard.

Previously (#1784), the upload UI for DE was functional, but had relatively sparse user orientation.  Now, the DE upload UI has visually-rich example data that conveys formatting requirements.  It also briefly explains the value of author-computed DE -- i.e. outlines for the study owner how uploading DE files makes their study more useful for researchers.  Finally, the "Differential expression" step is now first instead of last among the "Other files" steps, helping ensure users won't overlook it.

Along with #1844, this fit and finish builds on real user observations to improve onboarding for DE upload.

Here's how it looks!

### Before

<img width="1512" alt="Old_DE_upload_UI__SCP_2023-08-07" src="https://github.com/broadinstitute/single_cell_portal_core/assets/1334561/32a7d516-5bbc-4adb-9995-7ceb0c112566">

### After

<img width="1512" alt="New_DE_upload_UI__SCP_2023-08-08" src="https://github.com/broadinstitute/single_cell_portal_core/assets/1334561/3591e5de-ee88-4086-a2cc-da85c9468ce7">

### Test
* Go to upload wizard for a study, confirm no "Differential expression" tab shows by default (as it's not enabled)
* Set `show_de_upload` feature flag on the study, to enable DE upload
* Confirm "Differential expression" step is atop "Other files"
* Confirm "Differential expression" tab
* Confirm example data and other refinements described above are visible
* Confirm the table in "Wide format" scrolls horizontally

### Further detail
More context and discussion are available in [SCP demo video](https://drive.google.com/file/d/1Kaijvrk8aSXYQXkNHz7tWF2UFRATD9mn/view?t=12s).  This satisfies SCP-5235.

_Note_: you'll notice this PR changes some pre-existing draft code.  I inadvertently pushed that in my last PR.  If this PR doesn't look like it'll merge before the upcoming release, then I can open another PR to remove the draft.
